src/clr-angular/data/datagrid/providers/table-size.service.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {isPlatformBrowser} from "@angular/common";
+import {ElementRef, Inject, Injectable, PLATFORM_ID} from "@angular/core";
+
+/**
+ * @description
+ * Internal datagrid service that holds a reference to the clr-dg-table element and exposes a method to get height.
+ */
+@Injectable()
+export class TableSizeService {
+    private tableRef: Element;
+
+    constructor(@Inject(PLATFORM_ID) private platformId: Object) {}
+    public set table(table: ElementRef) {
+        if (isPlatformBrowser(this.platformId) && table.nativeElement) {
+            this.tableRef = table.nativeElement.querySelector(".datagrid-table");
```

Comment:
I just noticed that we actually don't need query DOM to find this table ref. The table service could be called inside `DatagridTableRenderer` and its `elementRef` is assigned to `TableSizeService.ref`. So we would no longer need to even inject this `PLATFORM_ID`.

Reply:
The table renderer is going away with this PR.